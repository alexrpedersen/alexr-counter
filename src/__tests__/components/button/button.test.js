import React from 'react';
import { shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Button from '../../../components/button/button.js';

describe('<Button />', () => {
    it('exists after my application has loaded', () => {
        let app = shallow(<Button />);
        expect(app.find('button').exists()).toBeTruthy();
        expect(app.find('button')).toHaveLength(2);
    });

    it('changes the state of our number on click -1', () =>{
        let app = mount(<Button />);
        let minusButton = app.find('button').at(0);
        minusButton.simulate('click');

        expect(app.state('count')).toEqual(-1);
    })

    it('changes the state of our number on click +1', () =>{
        let app = mount(<Button />);
        let plusButton = app.find('button').at(1);
        plusButton.simulate('click');

        expect(app.state('count')).toEqual(1);
    })

    it('renders correctly', () => {
        const tree = renderer.create(<Button />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});


