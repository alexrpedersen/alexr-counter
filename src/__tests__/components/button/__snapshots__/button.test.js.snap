// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<div
  className="Button"
>
  <div>
    <div
      className="count"
    >
      <h3>
        Count:
      </h3>
      <h1>
        0
      </h1>
    </div>
    <button
      onClick={[Function]}
    >
       - 
    </button>
    <button
      onClick={[Function]}
    >
       + 
    </button>
  </div>
</div>
`;
